src/DI/Compiler.php

```diff
@@ -106,13 +111,15 @@ public function processParameters()
 	/** @internal */
 	public function processExtensions()
 	{
-		for ($i = 0; $slice = array_slice($this->extensions, $i, 1, TRUE); $i++) {
-			$name = key($slice);
+		foreach ($this->extensions as $name => $extension) {
 			if (isset($this->config[$name])) {
 				$this->config[$name] = $tmp = Helpers::expand($this->config[$name], $this->builder->parameters);
 				unset($tmp['services']);
 				$this->extensions[$name]->setConfig($tmp);
 			}
+		}
+		for ($i = 0; $slice = array_slice($this->extensions, $i, 1, TRUE); $i++) {
+			$name = key($slice);
```

Comment:
It will work with most use cases. 
But it won't work with some edge cases - e.g. the test bellow. Config must be set right after [extension is added](https://github.com/nette/di/pull/50/files#diff-eaabdd91a6dedf0fa69456f1132d3483R22). Second edge case I can think of is when extension replaces other extension.